.editorconfig

```diff
@@ -0,0 +1,24 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+# brace expansion notation, does not work properly everywhere
+[*.h]
+indent_style = tab
+[*.cpp]
+indent_style = tab
+[*.txt]
+indent_size = 4
+[*.json]
+indent_size = 4
+[*.cmake]
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
```

Comment:
which editors care about this type of file?

Reply:
this thing makes sure that tools that understand editorconfig (quite a lot of them right now, including emacs, vim, atom, visual studio code, kate, qtcreator-editorconfig-plugin) will understand. if you don't like it I'll remove.